### PR TITLE
Instrumented Tests need to be executed for all supported API Levels

### PR DESCRIPTION
**Pull request by p3dr0rv**
## Currently instrumented tests are only running for API Level 30.  We need to run these nightly for all API levels.

We have created the next pipeline [build-publish-docker-image.yml](https://identitydivision.visualstudio.com/Engineering/_build?definitionId=1325&_a=summary) to build and publish [docker images](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/ddb6dea3-53f2-4b9f-b05c-1684e75ab420/resourceGroups/AuthClientAzureContainers/providers/Microsoft.ContainerRegistry/registries/AuthClient/repository) with different android API levels

The [instrumented-tests-multistage.yml](https://identitydivision.visualstudio.com/Engineering/_build?definitionId=1421)  pipeline runs everyday nightly  the Instrumented android Test on dev branch for every API level (30 to 21) and project (msal, adal, common) defined in the YAML.
image
You can run this pipeline manually and select the APIs to test, the projects and the branch to test 

image image

## Note
Android API level 27 cannot be build, still under investigation

### CODE REVIEW

**Comment by shahzaibj** [comment]
What about broker (AADAuthenticator) ?

**Review by iamgusain — APPROVED** [review]

File: azure-pipelines/instrumented-tests-multistage.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          acrRegistry: $(acrName).azurecr.io
+          acrRepository: samples/dbi-instrumented-api${{ api }}
+        displayName: 'test ${{ project }} API ${{ api }}'
+        steps:
+        - template: templates/docker-instrumented-test.yml
+          parameters:
+            acrImage: $(acrRegistry)/$(acrRepository)
+            androidProject: ${{ project }}
+            ${{ if eq(project, 'broker') }}:
+              EnvVstsMvnAccount: ENV_VSTS_MVN_ANDROIDADACCOUNTS_ACCESSTOKEN
```
Comment (shahzaibj) [inline azure-pipelines/instrumented-tests-multistage.yml:80]: nit: new line ending

**Review by shahzaibj — APPROVED** [review]

**Comment by shahzaibj** [comment]
Is there a way in the report to tell which test was run on which specific API level? It doesn't seem obvious

image

File: azure-pipelines/instrumented-tests-multistage.yml (hunk 1 above)
Comment (p3dr0rv) [reply azure-pipelines/instrumented-tests-multistage.yml]: yes Sr daf01617d63f67802890fadfbf7add8277ad9d23

**Comment by p3dr0rv** [comment]
> Is there a way in the report to tell which test was run on which specific API level? It doesn't seem obvious
> 
> image

Good catch !
what do you think of this view @shahzaibj 
image

**Comment by p3dr0rv** [comment]
https://identitydivision.visualstudio.com/Engineering/_build/results?buildId=788508&view=ms.vss-test-web.build-test-results-tab

Pipeline that test all the changes related to this issue

**Comment by shahzaibj** [comment]
> > Is there a way in the report to tell which test was run on which specific API level? It doesn't seem obvious
> > image
> 
> Good catch !
> what do you think of this view @shahzaibj
> image

@p3dr0rv I think this is good for now 👍

In the future we may want to improve this further by somehow trying to put the API level in each of the test names as well..something like this: 

image

It is up to you if you want to look into that right now or come back to it later in a separate PR 😉

**Pull request merged**